apim-operator/pkg/controller/api/api_controller.go

```diff
@@ -416,16 +414,63 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
 	}
 
 	analyticsEnabledBool, _ := strconv.ParseBool(analyticsEnabled)
-	dep := createMgwDeployment(instance, imageName, controlConf, analyticsEnabledBool, r)
+	dep := createMgwDeployment(instance, imageName, controlConf, analyticsEnabledBool, r, userNameSpace)
 	depFound := &appsv1.Deployment{}
 	deperr := r.client.Get(context.TODO(), types.NamespacedName{Name: dep.Name, Namespace: dep.Namespace}, depFound)
 
 	svc := createMgwLBService(instance, userNameSpace)
 	svcFound := &corev1.Service{}
 	svcErr := r.client.Get(context.TODO(), types.NamespacedName{Name: svc.Name, Namespace: svc.Namespace}, svcFound)
 
-	//Schedule Kaniko pod
-	if imageExist {
+	if instance.Spec.UpdateTimeStamp != "" {
+		//Schedule Kaniko pod
+		log.Info("Update API")
```

Comment:
Can we log which API we are updating here. It's better if the log is more descriptive

Reply:
Added the logs